### ISSUE TITLE
CRD scaffolds trigger false positives with kube-api-linter for Operator Projects

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kube-api-linter#192]
### Problem

Operator Projects scaffolds standard Kubernetes CRD patterns that kube-api-linter incorrectly flags as errors, even with `WhenRequired` configuration.

### What We're Scaffolding

```go
type MyResource struct {
    metav1.TypeMeta   `json:",inline"`
    metav1.ObjectMeta `json:"metadata,omitzero"`
    
    Spec   MyResourceSpec   `json:"spec"`
    Status MyResourceStatus `json:"status,omitzero"`
}

type MyResourceStatus struct {
    Conditions []metav1.Condition `json:"conditions,omitempty"`
}
```

**Error 1:**
```
optionalfields: field Status has a valid zero value ({}), but the validation 
is not complete (e.g. min properties/adding required fields). The field should 
be a pointer to allow the zero value to be set. If the zero value is not a 
valid use case, complete the validation and remove the pointer. (kubeapilinter)

    Status WordpressStatus `json:"status,omitzero"`
```
- Status is never a pointer in any Kubernetes API
- Works correctly with `omitzero` tag
- Validation incompleteness is expected - Status is controller-managed, not user-provided

**Error 2:**
```
requiredfields: field Spec has a valid zero value ({}), but the validation is 
not complete (e.g. min properties/adding required fields). The field should be 
a pointer to allow the zero value to be set. If the zero value is not a valid 
use case, complete the validation and remove the pointer. (kubeapilinter)

    Spec WordpressSpec `json:"spec"`
```
- Spec is never a pointer in Kubernetes APIs
- Scaffolds are starting points - users add validation when they implement their business logic

**Error 3:**
```
conditions: Conditions field in WordpressStatus is missing the following markers: 
patchStrategy=merge, patchMergeKey=type (kubeapilinter)

    Conditions []metav1.Condition `json:"conditions,omitempty"`
```
- `metav1.Condition` already handles patches correctly
- Adding these markers is redundant for this standard Kubernetes type

### What We Need

A way to configure the linter to understand CRD scaffolding patterns. Something like:

```yaml
lintersConfig:
  optionalfields:
    pointers:
      preference: WhenRequired
      skipCRDStatusFields: true  # Don't require pointers for Status in CRD types
  requiredfields:
    skipCRDSpecFields: true  # Don't require pointers for Spec in CRD types
  conditions:
    skipMetav1Condition: true  # metav1.Condition already has proper tags
```

Or even simpler - a preset for CRD scaffolds:

```yaml
lintersConfig:
  preset: "crd-scaffold"  # Use CRD-friendly defaults
```

**Comment by camilamacedo86** [comment]
/remove-lifecycle stale

We need to find sometime to address this one.

**Comment by camilamacedo86** [comment]
/remove-lifecycle stale

Could we freeze this one?
I do not think we can add that and start to promote before we have like a default profile for CRDs.

**Comment by JoelSpeed** [comment]
I don't think this issue belongs here. I think we need to come up with a solution on the kubebuilder side. The linter is correct, the issue is that the scaffolded types don't have valid examples of fields within the spec/status, so it treats them as having valid zero values and provides advice accordingly. Once users add fields, the advice will change based on their actual usage

**Comment by camilamacedo86** [comment]
Hi @JoelSpeed,

I understand that the linter rules for this project are based on the core Kubernetes API conventions, and that not all of them necessarily apply to projects that generate CRDs outside of the core Kubernetes API.

My suggestion would be to provide a dedicated profile for projects that generate CRDs but are not part of the core Kubernetes API.

Also, the issue is not only about `status`. The tutorial samples currently fail as well:

* Empty scaffold: https://github.com/kubernetes-sigs/kubebuilder/actions/runs/27908064952/job/82580189945?pr=4941
* Tutorial: https://github.com/kubernetes-sigs/kubebuilder/actions/runs/27908064952/job/82580189949?pr=4941

You can also see how people are configuring this on their side here: https://github.com/kubernetes-sigs/kubebuilder/pull/4941#discussion_r3457992961

The required configuration is quite verbose and exposes a lot of complexity to users. That makes for a poor UX, especially if Kubebuilder scaffolds a project that immediately needs this much configuration to pass the linter.

I think we should come up with a friendly abstraction that hides these details before promoting this more broadly.

However, if you find any issues with the scaffolds or tutorials, please let us know. That said, I don't think that's the case here, as we already went through those scenarios together and addressed them.

See the example shared in that comment: ( IHMO ask for someone do something like is not a good ux )

```yaml
 settings:
          linters:
            enable:
              # Ensures arrays of structs have at least one required field
              #
              # When an array contains structs where all fields are optional or unmarked, it becomes difficult to distinguish between different array elements in YAML configurations. This can lead to ambiguous or confusing API definitions.
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#arrayofstruct
              - "arrayofstruct"
              # Ensure comments start with the serialized version of the field name.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#commentstart
              - "commentstart"
              # Ensure conditions have the correct json tags and markers.
              #
              # Conditions are idiomatically the first field within the status struct, and the linter will highlight when the Conditions are not the first field.
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#conditions
              - "conditions"
              # Ensure mutually exclusive markers are not used on the same field
              #
              # See also the 'conditions' rule below with the proposed configuration for CRDs.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#conditions
              - "conflictingmarkers"
              # Ensure fields have either a default value or are required
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#defaultorrequired
              # see also the 'default_vs_required' conflicting markers rule below
              - "defaultorrequired"
              # Ensure default markers are configured correctly
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#defaults
              - "defaults"
              # Ensure there are no exact duplicate markers. for types and fields.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#duplicatemarkers
              - "duplicatemarkers"
              # Ensure only int32 and int64 are used for integers.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#integers
              - "integers"
              # Ensure every field has a json tag.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#jsontags
              - "jsontags"
              # Ensure all strings and arrays have maximum lengths/maximum items.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#maxlength
              - "maxlength"
              # Ensure all strings and arrays have minimum lengths/minimum items.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#minlength
              - "minlength"
              # Bools do not evolve over time, should use enums instead.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#nobools
              - "nobools"
              # Prevents usage of `Duration` types.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#nodurations
              - "nodurations"
              # Ensure floats are not used.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#nofloats
              - "nofloats"
              # Ensure maps are not used.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#nomaps
              - "nomaps"
              # Ensure that types and fields do not have the nullable marker.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#nonullable
              - "nonullable"
              # Prevents usage of 'Timestamp' fields
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#notimestamp
              - "notimestamp"
              # Ensure that all fields marked as optional adhere to being pointers and
              # having the `omitempty` value in their `json` tag where appropriate.
              #
              # Because of struct fields in the CRD, the pointer preference setting below is set to WhenRequired:
              # Linter will suggest to add the pointer to the field only when the field zero value is a valid value
              # for the field. When the field zero value is not a valid value for the field, the linter will suggest
              # to remove the pointer from the field. When the field zero value is not a valid value for the field
              # of type struct, the linter will suggest to add omitzero json tag and to remove the pointer from the field.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#optionalfields
              - "optionalfields"
              # Ensure that every field should be marked as `+optional` or `+required`.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#optionalorrequired
              - "optionalorrequired"
              # Ensure that required fields should not be pointers, and should not have `omitempty`.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#requiredfields
              - "requiredfields"
              # Ensure that all array fields have the appropriate listType markers.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#ssatags
              - "ssatags"
              # Ensure that root objects that have a `status` field should have a status subresource.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#statussubresource
              - "statussubresource"
              # Ensure that types and fields do not contain more than a single definition of a marker that should only be present once.
              #
              # xref: https://github.com/kubernetes-sigs/kube-api-linter/blob/main/docs/linters.md#uniquemarkers
              - "uniquemarkers"
            disable:
              - "*"
          lintersConfig:
            conflictingmarkers:
              conflicts:
                - name: "default_vs_required"
                  sets:
                    - ["default", "kubebuilder:default"]
                    - [
                        "required",
                        "kubebuilder:validation:Required",
                        "k8s:required",
                      ]
                  description: "A field with a default value cannot be required"
            conditions:
              isFirstField: Warn # Conditions are idiomatically the first field within the status struct, and the linter will highlight when the Conditions are not the first field.
              useProtobuf: Forbid # For CRDs, protobuf tags are not necessary.
              usePatchStrategy: Forbid # For CRDs, patchStrategy is not necessary.
            optionalFields:
              pointers:
                preference: WhenRequired
```

**Comment by camilamacedo86** [comment]
Give me some time like 2/3 weeks I will try to find some time to look that with the love. 💌  to see if I can came out with something or an proposal.
